src/repertoire/config.py

```diff
@@ -28,6 +37,67 @@ class SentryConfig(BaseModel):
     enabled: bool = Field(False, title="Whether to send exceptions to Sentry")
 
 
+class TapServerConfig(BaseModel):
+    """Configuration for a TAP server."""
+
+    model_config = ConfigDict(
+        alias_generator=to_camel,
+        extra="forbid",
+        populate_by_name=True,
+    )
+
+    enabled: bool = Field(
+        True,
+        title="Enable TAP schema management for this server",
+        description=(
+            "Controls whether repertoire manages TAP_SCHEMA for this server. "
+            "Set to false in environments where this TAP server doesn't exist "
+            "or doesn't use CloudSQL."
+        ),
+    )
+
+    schema_version: str | None = Field(
+        None,
+        title="Schema version override for this server",
+        description=(
+            "Override the global schemaVersion for this specific server. "
+            "If not set, uses config.schemaVersion."
+        ),
+    )
+
+    schemas: list[str] = Field(
+        ...,
+        title="List of schema names to load",
+        description="Schema YAML file names (without .yaml extension)",
+    )
+
+    database: str = Field(
+        ...,
+        title="Database name",
+        description="Database name for this TAP server",
+    )
+
+    database_user: str = Field(
+        ...,
+        title="Database username",
+        description="Username for this TAP server",
+    )
+
+    database_password_key: str = Field(
+        ...,
+        title="Secret key for database password",
+        description="Key in repertoire secret containing the server password",
+    )
+
+    @field_validator("schemas")
+    @classmethod
+    def schemas_not_empty(cls, v: list[str]) -> list[str]:
+        """Ensure schemas list is not empty."""
+        if not v:
+            raise ValueError("schemas list cannot be empty")
+        return v
```

Comment:
You can just pass `min_length=1` in the arguments to `Field` for `schemas` and then don't need this validator.

Reply:
Oh right I forgot that this was possible good idea!